core/modules/layout/plugins/context/layout_context.inc

```diff
@@ -269,6 +269,14 @@ abstract class LayoutContext extends LayoutHandler {
    * The summary is displayed in the context table on Layout settings pages.
    */
   function getAdminSummary() {}
+
+   /** 
+   * Set the context data from the router item array.
+   * 
+   * @param array $router_item
+   *   A router item array.
+   */
+  function setDataFromRouterItem($router_item) {}
```

Comment:
Can you move this up under `setData()` so the two methods are next to each other?

Reply:
Done